framework/src/wifi_manager/wifi_manager_cb.c

```diff
@@ -110,40 +110,58 @@ void _handle_user_cb(_wifimgr_usr_cb_type_e evt, void *arg)
 		switch (evt) {
 		case CB_STA_CONNECTED:
 			WM_LOG_INFO("[WM] call sta connect success event\n");
-			cbk->sta_connected(WIFI_MANAGER_SUCCESS);
+			if (cbk->sta_connected) {
+				cbk->sta_connected(WIFI_MANAGER_SUCCESS);
+			}
 			break;
 		case CB_STA_CONNECT_FAILED:
 			WM_LOG_INFO("[WM] call sta connect fail event\n");
 			WIFIADD_ERR_RECORD(ERR_WIFIMGR_CONNECT_FAIL);
-			cbk->sta_connected(WIFI_MANAGER_FAIL);
+			if (cbk->sta_connected) {
+				cbk->sta_connected(WIFI_MANAGER_FAIL);
+			}
 			break;
 		case CB_STA_DISCONNECTED:
 			WM_LOG_INFO("[WM] call sta disconnect event\n");
-			cbk->sta_disconnected(WIFI_MANAGER_DISCONNECT);
+			if (cbk->sta_disconnected) {
+				cbk->sta_disconnected(WIFI_MANAGER_DISCONNECT);
+			}
 			break;
 		case CB_STA_RECONNECTED:
 			WM_LOG_INFO("[WM] call sta reconnect event\n");
-			cbk->sta_disconnected(WIFI_MANAGER_RECONNECT);
+			if (cbk->sta_disconnected) {
+				cbk->sta_disconnected(WIFI_MANAGER_RECONNECT);
+			}
 			break;
 		case CB_STA_JOINED:
 			WM_LOG_INFO("[WM] call sta join event\n");
-			cbk->softap_sta_joined();
+			if (cbk->softap_sta_joined) {
+				cbk->softap_sta_joined();
+			}
 			break;
 		case CB_STA_LEFT:
 			WM_LOG_INFO("[WM] call sta leave event\n");
-			cbk->softap_sta_left();
+			if (cbk->softap_sta_left) {
+				cbk->softap_sta_left();
+			}
 			break;
 		case CB_SCAN_DONE:
 			WM_LOG_INFO("[WM] call sta scan event\n");
 			/* convert scan data.*/
 			wifi_manager_scan_info_s *info = NULL;
 			wifi_utils_scan_list_s *list = (wifi_utils_scan_list_s *)arg;
 			if (list) {
-				WIFIMGR_CHECK_RESULT_CLEANUP(
-					_convert_scan_info(&info, (wifi_utils_scan_list_s *)arg),
-					"parse error", , cbk->scan_ap_done(NULL, WIFI_SCAN_FAIL));
-				cbk->scan_ap_done(&info, WIFI_SCAN_SUCCESS);
-				_free_scan_info(info);
+				if (WIFI_MANAGER_SUCCESS != _convert_scan_info(&info, (wifi_utils_scan_list_s *)arg)) {
```

Comment:
how about this
///
    if (list) {
        wifi_manager_scan_result_e result = WIFI_MANAGER_SUCCESS;
        if (WIFI_MANAGER_SUCCESS != _convert_scan_info(&info, (wifi_utils_scan_list_s *)arg)) {
            WM_LOG_ERROR("[WM] parse error\n");           
            result = result = WIFI_SCAN_FAIL;
        }
         
        if (cbk->scan_ap_done) {
            cbk->scan_ap_done(&info, WIFI_SCAN_SUCCESS);
        }
        
        _free_scan_info(info);
    }
///

because of,